### PR TITLE
[WIP] Fix bounce mail handling

### PR DESCRIPTION
**Pull request by pixelbrackets**
I tested the patch by @linkpool (ZIP file in issue #102) and was able to fetch mails with it. Therefore I pasted this into a commit and create this PR so that you may take a look at the code changes.

### CODE REVIEW

File: Classes/Scheduler/AnalyzeBounceMail.php [hunk 1]
```diff
@@ -211,8 +211,8 @@ private function processBounceMail($message)
                 $bouncedMail = $attachment->getData();
                 // Find mail id
                 $midArray = $readMail->find_XTypo3MID($bouncedMail);
-                if (is_array($midArray)) {
-                    // if mid, rid and rtbl are found, then continue
+                if (false === empty($midArray)) {
```
Comment (sgrossberndt) [inline Classes/Scheduler/AnalyzeBounceMail.php]: if (!empty($midArray))

File: Resources/Private/Php/Fetch/src/Fetch/Message.php [hunk 2]
```diff
@@ -543,7 +543,7 @@ protected function processStructure($structure, $partIdentifier = null)
                 }
             }
 
-            if (strtolower($structure->subtype) === 'plain' || ($structure->type == 1 && strtolower($structure->subtype) !== 'alternative')) {
+            if ((strtolower($structure->subtype) === 'plain') || ($structure->type == 1 && strtolower($structure->subtype) !== 'alternative') || ($structure->type == 0 && strtolower($structure->subtype) == 'rfc822-headers')) {
```
Comment (sgrossberndt) [inline Resources/Private/Php/Fetch/src/Fetch/Message.php]: The Fetch sourcecode which is used to read the email has its own repo at https://github.com/tedious/Fetch/blob/master/src/Fetch/Message.php#L546

Although there seems not to be much activity in that project the change to Fetch should be made upstream by adding a pull request there.

If its not merged promptly (and Fetch can be updated here) at least a comment this is a modified version should be added

**Comment by typecat** [comment]
How is the status here? Is here something that needs to be done that prevents the merge?

**Comment by sgrossberndt** [comment]
There is no according pull request in https://github.com/tedious/Fetch/pulls yet - but there seems to be no activiry at all anymore. Either one of the forks in https://github.com/tedious/Fetch/network/members should be selected for this project or the code just fixed in direct_mail directly..

**Pull request merged**